### ISSUE TITLE
SanityImageEntity error triggered on gatsby-build for Gatsby v4

### ISSUE THREAD

**Issue by ajmalafif** [coreyward/gatsby-plugin-sanity-image#19]
I am a bit conscious before posting this here because this might not be `gatsby-plugin-sanity-image` specific issue since from the past I have @coreyward has always been on the forefront and proactive in bridging Gatsby gaps with Sanity.

**Context**
I am seeing these error when I ran `gatsby build` for [this repo](https://github.com/ajmalafif/ajmalafif.com/tree/gatsby-v4)
```
There was an error in your GraphQL query:

Unknown type "SanityImageEntity". Did you mean "SanityImageFit", "SanityImageAsset", "SanityImage", "SanityImageCrop",
 or "SanityImageHotspot"?

   1 | fragment GatsbyImageSharpFluid on ImageSharpFluid {
   2 |   base64
   3 |   aspectRatio
   4 |   src
   5 |   srcSet
   6 |   sizes
   7 | }
   8 |
>  9 | fragment ImageWithPreview on SanityImageEntity {
     |                              ^
  10 |   ...Image
  11 |   asset {
  12 |     metadata {
  13 |       preview: lqip
  14 |     }
  15 |   }
  16 | }
  17 |
  18 | fragment Image on SanityImageEntity {
  19 |   asset {

File path: /Users/minimaru/Sites/ajmalafif.com/web/src/templates/blog-post.js
Url path: /notes/learning-in-public/
Plugin: none
```
So when I am investigating and searching for `SanityImageEntity` points me to a `node_modules` folder for `gatsby-plugin-sanity-image`:

![CleanShot 2021-10-28 at 12 51 37@2x](https://user-images.githubusercontent.com/827167/139189175-2df1b360-cb0b-4a56-9498-4689d0eee473.jpg)

I apologize in advanced if this is out of scope or my lack of understanding. Appreciate any pointers and guidance if its relevant else feel free to close the issue! Thanks again!

**Comment by ajmalafif** [comment]
Is there a way to go around `SanityImageEntity` or its required that I use it?

**Comment by coreyward** [comment]
You can specify a different `fragmentTypeName` in your config if you'd like (anything other than `SanityImage` that doesn't conflict with another type in your app), but I don't think it will change the issue you're having. From your original post, you are showing a file that has different fragments in it than what is in gatsby-plugin-sanity-image:

<img width="498" alt="Screen Shot 2021-10-31 at 6 33 51 PM" src="https://user-images.githubusercontent.com/81224/139604939-da964db4-0c6f-4695-8f17-11e7e0e1eb62.png">

You might have a local file that is .gitignore'd or have something cached strange. Deleting `node_modules` and re-installing dependencies might help if so.

**Comment by coreyward** [comment]
This should be fixed in `gatsby-plugin-sanity-image@10.0.0`. See #20 for more detail.
